mesatee_services/fns/sgx_trusted_lib/src/trusted_worker/naive_bayes.rs

```diff
@@ -0,0 +1,154 @@
+use rusty_machine::learning::SupModel;
+use rusty_machine::linalg::Matrix;
+
+use serde_derive::Deserialize;
+use serde_json;
+
+#[derive(Deserialize)]
+pub(crate) struct NaiveBayesPayload {
+    input_model_columns: usize,
+    input_model_data: String,
+    target_model_columns: usize,
+    target_model_data: String,
+}
+
+pub struct NaiveBayesWorker {
+    worker_id: u32,
+    func_name: String,
+    func_type: FunctionType,
+    input: Option<NaiveBayesInput>,
+}
+
+struct NaiveBayesInput {
+    /// Sample model data
+    input_model_data: Matrix<f64>,
+    /// The target(also called label or result) of the sample model data
+    target_model_data: Matrix<f64>,
+}
+
+impl NaiveBayesWorker {
+    pub fn new() -> Self {
+        NaiveBayesWorker {
+            worker_id: 0,
+            func_name: "naive_bayes".to_string(),
+            func_type: FunctionType::Single,
+            input: None,
+        }
+    }
+}
+
+impl Worker for NaiveBayesWorker {
+    fn function_name(&self) -> &str {
+        self.func_name.as_str()
+    }
+    fn function_type(&self) -> FunctionType {
+        self.func_type
+    }
+    fn set_id(&mut self, worker_id: u32) {
+        self.worker_id = worker_id;
+    }
+    fn id(&self) -> u32 {
+        self.worker_id
+    }
+    fn prepare_input(
+        &mut self,
+        dynamic_input: Option<String>,
+        _file_ids: Vec<String>,
+    ) -> Result<()> {
+        let payload = dynamic_input.ok_or_else(|| Error::from(ErrorKind::MissingValue))?;
+
+        let naive_bayes_reg_payload: NaiveBayesPayload = serde_json::from_str(&payload)
+            .or_else(|_| Err(Error::from(ErrorKind::InvalidInputError)))?;
+
+        let input = parse_input_to_matrix(
+            &naive_bayes_reg_payload.input_model_data,
+            naive_bayes_reg_payload.input_model_columns,
+        )?;
+
+        let target = parse_input_to_matrix(
+            &naive_bayes_reg_payload.target_model_data,
+            naive_bayes_reg_payload.target_model_columns,
+        )?;
+        self.input = Some(NaiveBayesInput {
+            input_model_data: input,
+            target_model_data: target,
+        });
+        Ok(())
+    }
+
+    fn execute(&mut self, _context: WorkerContext) -> Result<String> {
+        let input = self
+            .input
+            .take()
+            .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
+        // Create a Gaussian Naive Bayes classifier.
+        let mut model = NaiveBayes::<Gaussian>::new();
+        // Train the model
+        model
+            .train(&input.input_model_data, &input.target_model_data)
+            .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+        // Predict the classes on the input data
+        let predict = model
+            .predict(&input.input_model_data)
+            .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+
+        let mut output = String::new();
+        writeln!(&mut output, "{:?}", predict)
```

Comment:
The output would be ```Matrix { rows: 6, cols: 3, data: [1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0] }```
Can you format the result, not directly return the debug output?